sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -193,106 +194,264 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
 
 ### HttpRequest and HttpResponse
 
-The HttpRequest and HttpResponse objects represent a generic concept of HTTP request and response constructs and are in no way tied to a particular transport or HTTP library.
+The `HttpRequest` and `HttpResponse` objects represent a generic concept of HTTP request and response constructs and are in no way tied to a particular transport or HTTP library.
 
-The HttpRequest has the following API. It does not vary between transports:
+The `HttpRequest` has the following API. It does not vary between transports:
 
 ```python
-class HttpRequest(object):
-
-    def __init__(self, method, url, headers=None, files=None, data=None):
-        self.method = method
+class HttpRequest:
+
+    def __init__(
+        self,
+        method: str,
+        url: str,
+        *,
+        params: Optional[ParamsType] = None,
+        headers: Optional[MutableMapping[str, str]] = None,
+        json: Any = None,
+        content: Optional[ContentType] = None,
+        data: Optional[dict] = None,
+        files: Optional[FilesType] = None,
+        **kwargs
+    ):
         self.url = url
-        self.headers = CaseInsensitiveDict(headers)
-        self.files = files
-        self.data = data
+        self.method = method
+        self.headers = CaseInsensitiveDict(default_headers)
+
+    @property
+    def content(self) -> Any:
+        """Get's the request's content
+
+        :return: The request's content
```

Comment:
It is already showed in def content(self) -> Any:.

For dev doc, I don't think we need to duplicate the information?